### PR TITLE
RobotPoseEstimator Enhancements

### PR DESCRIPTION
**Pull request by hadley31**
Update `RobotPoseEstimator` class to accept a `List` object instead of `ArrayList` to allow for flexible user implementation.

### CODE REVIEW

**Comment by hadley31** [comment]
Closes #678

File: photon-lib/src/main/java/org/photonvision/RobotPoseEstimator.java [hunk 1]
```diff
@@ -93,37 +97,37 @@ public RobotPoseEstimator(
...
-            return Pair.of(lastPose, 0.);
+            return Optional.empty();
         }
         Pair<Pose3d, Double> pair;
         switch (strategy) {
             case LOWEST_AMBIGUITY:
                 pair = lowestAmbiguityStrategy();
                 lastPose = pair.getFirst();
-                return pair;
+                return Optional.of(pair);
```
Comment (mcm001) [inline photon-lib/src/main/java/org/photonvision/RobotPoseEstimator.java:110]: Nit: I bet you could initialize the pair to null at the top of the switch-case, and have the default-case break. Then, you only need one `return optional.ofNullable(pair)` at the bottom.

**Review by mcm001 — DISMISSED** [review]

File: photon-lib/src/main/java/org/photonvision/RobotPoseEstimator.java (hunk 1 above)
Comment (hadley31) [reply photon-lib/src/main/java/org/photonvision/RobotPoseEstimator.java:110]: Was thinking that too 🙂 Figured I'd make the simplest change first. I will update that

**Review by gerth2 — APPROVED** [review]
Looks pretty solid. I'm not an expert in some of the language features used here but it looks directionally correct. Since the testcases are still passing, I'm fine with merging it. I'll try picking it up later this week into an example proejct

**Pull request merged**